src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -11397,6 +11775,76 @@ namespace HVACUnitarySystem {
 	}
 
 	Real64
+	TESIceStorageCoilOutletResidual(
+		Real64 PartLoadRatio, // compressor cycling ratio (1.0 is continuous, 0.0 is off)
+		Array1< Real64 > const & Par // data array
+	)
+	{
+
+		// FUNCTION INFORMATION:
+		//       AUTHOR        Richard Raustad, FSEC
+		//       DATE WRITTEN   August 2015
+		//       MODIFIED
+		//       RE-ENGINEERED
+
+		// PURPOSE OF THIS FUNCTION:
+		// Calculates residual function (desired outlet temp - actual outlet temp)
+		// Coil output depends on the part load ratio which is being varied to zero the residual.
+
+		// METHODOLOGY EMPLOYED:
+		// Calls SimWatertoAirHP or SimWatertoAirHPSimple to get outlet humidity ratio at the given cycling ratio
+		// and calculates the residual as defined above
+
+		// REFERENCES:
+
+		// Using/Aliasing
+		using PackagedThermalStorageCoil::SimTESCoil;
+
+		// Return value
+		Real64 Residuum; // residual to be minimized to zero
+
+		// Argument array dimensioning
+
+		// Locals
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+		// Par( 1 ) = double( UnitarySysNum );
+		// Par( 2 ) = DesOutTemp;
+		// Par( 3 ) = 0.0; // DesOutHumRat; set to 0 if temp controlled
+
+		// FUNCTION PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS
+		// na
+
+		// DERIVED TYPE DEFINITIONS
+		// na
+
+		// FUNCTION LOCAL VARIABLE DECLARATIONS:
+		int UnitarySysNum; // index of this coil
+		Real64 DesiredOutletTemp; // temperature control point
+		Real64 DesiredOutletHumRat; // humidity ratio control point
+		Real64 OutletAirTemp; // outlet air temperature [C]
+		Real64 OutletAirHumRat; // outlet air humidity ratio [kg/kg]
+
+		UnitarySysNum = int( Par( 1 ) );
+		DesiredOutletTemp = Par( 2 );
+		DesiredOutletHumRat = Par( 3 );
+
+		SimTESCoil( UnitarySystem( UnitarySysNum ).CoolingCoilName, UnitarySystem( UnitarySysNum ).CoolingCoilIndex, UnitarySystem( UnitarySysNum ).FanOpMode, UnitarySystem( UnitarySysNum ).TESOpMode, PartLoadRatio );
+
+		if ( DesiredOutletHumRat > 0.0 ) {
+			OutletAirHumRat = Node( UnitarySystem( UnitarySysNum ).CoolCoilOutletNodeNum ).HumRat;
+			Residuum = OutletAirHumRat - DesiredOutletHumRat;
```

Comment:
wondering if this will converge well with a tolerance of 0.001 since the residual is the difference between humidity ratios.